internal/plugin/validators/lua_validator.go

```diff
@@ -108,16 +108,26 @@ func (v *LuaValidator) ValidateSchema(ctx context.Context, pluginSchema string)
 	return pluginName, nil
 }
 
-func validationErr(name string, e error) error {
+func (v *LuaValidator) validationErr(name string, e error) error {
 	if e == nil {
 		return nil
 	}
+	res := validation.Error{}
 	var errMap map[string]interface{}
 	err := json.ProtoJSONUnmarshal([]byte(e.Error()), &errMap)
 	if err != nil {
-		return fmt.Errorf("unmarshal kong plugin validation error: %v", err)
+		v.logger.With(zap.String("plugin", name)).Error("validation error", zap.Error(e))
+		res.Errs = append(res.Errs, &grpcModel.ErrorDetail{
```

Comment:
Can we throw in a quick unit test for this?

Reply:
I was tempted to add one for the one case I see this failing, but that should be fixed soon, so the test would be thrown away :\ 

Any suggestion on how to consistently replicate one?

Reply:
Yeah, honestly I wouldn't sweat too much here, due to the nature of this. But sure, let me know if something comes to mind!

Reply:
This was a _lot_ harder than it should have been given the drastic use of proper interfaces and exposing interfaces instead of concrete types, but there you go!

```diff
diff --git a/internal/plugin/validators/badtestdata/embed.go b/internal/plugin/validators/badtestdata/embed.go
new file mode 100644
index 0000000..820e1a4
--- /dev/null
+++ b/internal/plugin/validators/badtestdata/embed.go
@@ -0,0 +1,8 @@
+package badtestdata
+
+import (
+	"embed"
+)
+
+//go:embed lua-tree/share/lua/5.1/*
+var BadLuaTree embed.FS
diff --git a/internal/plugin/validators/badtestdata/lua-tree/share/lua/5.1/validate-invalid-json.lua b/internal/plugin/validators/badtestdata/lua-tree/share/lua/5.1/validate-invalid-json.lua
new file mode 100644
index 0000000..11d6aea
--- /dev/null
+++ b/internal/plugin/validators/badtestdata/lua-tree/share/lua/5.1/validate-invalid-json.lua
@@ -0,0 +1,3 @@
+_G["validate"] = function()
+  return nil, "this is invalid JSON"
+end
diff --git a/internal/plugin/validators/lua_validator_test.go b/internal/plugin/validators/lua_validator_test.go
index 102b127..d1bce64 100644
--- a/internal/plugin/validators/lua_validator_test.go
+++ b/internal/plugin/validators/lua_validator_test.go
@@ -18,6 +18,7 @@ import (
 	"github.com/kong/koko/internal/model/json/validation"
 	"github.com/kong/koko/internal/model/json/validation/typedefs"
 	"github.com/kong/koko/internal/plugin"
+	"github.com/kong/koko/internal/plugin/validators/badtestdata"
 	"github.com/kong/koko/internal/plugin/validators/testdata"
 	"github.com/kong/koko/internal/resource"
 	serverUtil "github.com/kong/koko/internal/server/util"
@@ -265,6 +266,30 @@ func TestProcessAutoFields(t *testing.T) {
 
 func TestValidate(t *testing.T) {
 	validator := goodValidator
+	t.Run("invalid JSON returned", func(t *testing.T) {
+		badValidator, err := NewLuaValidator(Opts{
+			Logger:   log.Logger,
+			InjectFS: &badtestdata.BadLuaTree,
+		})
+		require.NoError(t, err)
+		require.NoError(t, badValidator.LoadSchemasFromEmbed(plugin.Schemas, "schemas"))
+		require.NoError(t, badValidator.LoadPatch("validate-invalid-json"))
+
+		err = badValidator.Validate(context.Background(), &grpcModel.Plugin{
+			Name:      "prometheus",
+			Protocols: []string{"http", "https"},
+			Enabled:   wrapperspb.Bool(true),
+		})
+		require.IsType(t, validation.Error{}, err)
+		vErr := err.(validation.Error)
+		require.Len(t, vErr.Errs, 1)
+		require.Len(t, vErr.Errs[0].Messages, 1)
+		require.Equal(
+			t,
+			"(prometheus) unknown plugin validation error, please file a bug with Kong Inc",
+			vErr.Errs[0].Messages[0],
+		)
+	})
 	t.Run("test with entity errors", func(t *testing.T) {
 		config, err := structpb.NewStruct(map[string]interface{}{
 			"policy": "redis",
```

Reply:
This is great man! Please feel free to commit and push it directly here, happy to co-author this PR! :)

Reply:
Pushed it up in [261ef79](https://github.com/Kong/koko/pull/463/commits/261ef79baacdd587e95012fc4f3c61f89aeef047) 🙂

Reply:
Awesome, thanks a lot! :)